tests/ignite/metrics/test_fbeta.py

```diff
@@ -71,19 +99,31 @@ def update_fn(engine, batch):
         y_true_batch = next(y_true_batch_values)
         y_pred_batch = next(y_pred_batch_values)
         if output_transform is not None:
-            return {"y_pred": torch.from_numpy(y_pred_batch), "y": torch.from_numpy(y_true_batch)}
-        return torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+            return {"y_pred": y_pred_batch, "y": y_true_batch}
+        return y_pred_batch, y_true_batch
 
     evaluator = Engine(update_fn)
 
-    f2 = Fbeta(beta=2.0, average=average, precision=p, recall=r, output_transform=output_transform)
+    device = None if p is not None and r is not None else available_device
+    f2 = Fbeta(beta=2.0, average=average, precision=p, recall=r, output_transform=output_transform, device=device)
+
     f2.attach(evaluator, "f2")
 
     data = list(range(n_iters))
     state = evaluator.run(data, max_epochs=1)
 
-    f2_true = fbeta_score(y_true, np.argmax(y_pred, axis=-1), average="macro" if average else None, beta=2.0)
-    np.testing.assert_allclose(np.array(f2_true), np.array(state.metrics["f2"]))
+    f2_pred = to_numpy_float32(state.metrics["f2"])
```

Comment:
Why we should convert to f32 for all devices: cpu, cuda, mps?

Reply:
great observation. But to be honest, I don't see how to make conversions only on MPS without overly polluting the code with device checks. How would you like to do that?

Reply:
Fbeta should handle dtype internally, for cpu and cuda as f64 is available, the output should be f64 (to verify). For mps, precision/recall value should be f32. So, to compare output with the expected we can compare float values directly and have to convert output to numpy array and it should keep the dtype from torch tensor. What do you think?